### PR TITLE
registry views for LMO, GENE and ORGA

### PR DESCRIPTION
**Pull request by tahmeedshah**
show count in registries main page
registry views for LMO, GENE and ORGA
CSS change for search hover-effect, margin between child and parent component issue is in bch-header.html

### CODE REVIEW

File: app/views/reports/bch/registries/index.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+                        query: 'schema_s:(' + schemas.join(' ') + ')'
+                    };
+				 
+					return searchService.facets(recordQuery)
+						.then(function (result) { 
+							$scope.count = result.schema_s;
+						})
+					.catch(function(e){
+						toastr.error('There was an error running search query.')
+						var exception = {
```
Comment (blaisefonseca) [inline app/views/reports/bch/registries/index.js:20]: put the error message in html 
`Error loading registry records, please try again.`

File: app/views/reports/bch/registries/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+							data    :  e.data||e.message, status:e.status,
+							url     : (e.config||{}).url, params: (e.config||{}).params,
+							stack   : e.stack
+						}                                
+						$log.error(JSON.stringify(exception))
```
Comment (blaisefonseca) [inline app/views/reports/bch/registries/index.js:25]: you can just throw e, should be taken care by the global handler

File: app/views/reports/bch/registries/view-gene-registry.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+					stack   : e.stack
+				}                                
+				$log.error(JSON.stringify(exception))
+			})
+			.finally(function(){
+				$scope.isLoading = false;
+			})					  
+	}
+	$scope.export = function () {
+		$scope.dataToExport = $scope.data;
```
Comment (blaisefonseca) [inline app/views/reports/bch/registries/view-gene-registry.js:38]: what is the use of `dataToExport `

File: app/views/reports/bch/registries/view-orga-registry.html [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+								<th class="display-20">Taxonomic status</th>
+								<th class="display-17">Common name</th>
+								<th class="display-45">Description</th>
+							</tr>
+						</thead>
+						<tbody>
+							<tr ng-repeat="row in ::orgaRecords">
+								<td class="tableexport-string">
+									<a rel="noopener" translation-url target="_blank" ng-if="::row.url_ss.length" rel="noopener" href="{{::row.url_ss[0]}}">
+										<span class="text-uppercase" ng-bind="::row.recordId"></span>
```
Comment (blaisefonseca) [inline app/views/reports/bch/registries/view-orga-registry.html:49]: use ng-href

File: app/views/reports/bch/registries/view-gene-registry.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+app.controller("geneRegistryController", ['$scope','searchService','$element', '$timeout', 'toastr', '$log',
+function($scope,searchService,$element,$timeout, toastr, $log) { 
+		$scope.isLoading = false;
+		loadRecords();
+		function loadRecords(){ 
+			$scope.isLoading = true;
+		var searchQuery = {
+			fields:  'recordId:uniqueIdentifier_s,name:name_s,trait:summary_t,donorOrganism:geneDonorOrganism,geneFunction:biologicalFunction_EN_t,url_ss',
+			query:  'schema_s:dnaSequence',
```
Comment (blaisefonseca) [inline app/views/reports/bch/registries/view-gene-registry.js:11]: wrong field name geneDonorOrganism

**Review by blaisefonseca — CHANGES_REQUESTED** [review]
please test download as well

File: app/css/registry.css [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+.tableexport-caption{
+	display: none;       
+}
+.registry-details thead th {background-color: #f5f5f5;color: #333;}
+.registry-details .export-btn{margin-right: 5px; margin-bottom: 5px;}
+.registry-details .list-group-item{margin-bottom: 5px;}
+.registry-details .list-group-item:first-child{color: #ffffff;}
+.registry-details .p-borders{padding-top:10px;border-bottom: 2px solid #eee;padding-bottom:10px;}
+.registry-details .display-15 {width: 15%;}
+.registry-details .display-15 {width: 15%;}
```
Comment (blaisefonseca) [inline app/css/registry.css:17]: remove duplicate css

File: app/views/reports/bch/registries/index.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+		'./index'], function(app, _) { 'use strict';
+
+		app.controller("registryReportController", ['$scope', 'searchService', 'toastr', '$log',
+			function ($scope, searchService, toastr, $log) {
+				var schemas = ["organism", "modifiedOrganism", "dnaSequence"];
+				loadRecords();
+				function loadRecords(schema) {
+					var recordQuery = { 
+                        fields: 'schema_s',
+                        query: 'schema_s:(' + schemas.join(' ') + ')'
```
[the remark this answers is not in the document]
Comment (blaisefonseca) [reply app/views/reports/bch/registries/index.js:11]: _.map(schemas, solr.escape).join(', ')

File: app/views/reports/bch/registries/index.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
+define(['app', 'lodash', 'css!/app/css/registry.css', 'services/search-service', 'js/common','views/register/directives/register-top-menu',
+		'./index'], function(app, _) { 'use strict';
```
[the remark this answers is not in the document]
Comment (tahmeedshah) [reply app/views/reports/bch/registries/index.js:2]: "views/register/directives/register-top-menu" is required to show top menu of "Dashboard | Requests| Alerts | User Management | Preferences | Reports"

**Pull request closed without merging**